controllers/providers/assistant/gslb.go

```diff
@@ -60,27 +61,47 @@ func NewGslbAssistant(client client.Client, k8gbNamespace, edgeDNSServer string)
 
 // CoreDNSExposedIPs retrieves list of IP's exposed by CoreDNS
 func (r *GslbLoggerAssistant) CoreDNSExposedIPs() ([]string, error) {
+	serviceList := &corev1.ServiceList{}
 	coreDNSService := &corev1.Service{}
-	err := r.client.Get(context.TODO(),
-		types.NamespacedName{Namespace: r.k8gbNamespace, Name: coreDNSExtServiceName}, coreDNSService)
+	sel, err := labels.Parse(coreDNSServiceLabel)
+	if err != nil {
+		log.Error().Err(err).Msg("Badly formed label selector")
+		return nil, err
+	}
+	listOption := &client.ListOptions{
+		LabelSelector: sel,
+		Namespace:     r.k8gbNamespace,
+	}
+
+	err = r.client.List(context.TODO(), serviceList, listOption)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			log.Warn().Msgf("Can't find %s service", coreDNSExtServiceName)
+			log.Warn().Err(err).Msg("Can't find CoreDNS service")
+		}
+	}
+	if len(serviceList.Items) != 1 {
+		log.Info().Msgf("items are not 1 actual %d", len(serviceList.Items))
+		for _, service := range serviceList.Items {
+			log.Warn().Msgf("found service %s", service.Name)
 		}
+		err := coreerrors.New("can't find CoreDNS service")
 		return nil, err
 	}
+	coreDNSService = &serviceList.Items[0]
+
 	var lbHostname string
 	if len(coreDNSService.Status.LoadBalancer.Ingress) > 0 {
```

Comment:
Can there be a case where service has no Status , or Status hase no LoadBalancer?

Reply:
@kuritka  this is for service type ClusterIP:
```yaml
status:
  loadBalancer: {}
```